src/spikeinterface/core/base.py

```diff
@@ -1006,38 +995,18 @@ def save_to_zarr(
                 folder = Path(folder)
                 if folder.suffix != ".zarr":
                     folder = folder.parent / f"{folder.stem}.zarr"
+                if folder.is_dir() and overwrite:
+                    shutil.rmtree(folder)
                 zarr_path = folder
-                zarr_path_init = str(zarr_path)
             else:
                 zarr_path = folder
-                zarr_path_init = zarr_path
 
         if isinstance(zarr_path, Path):
             assert not zarr_path.exists(), f"Path {zarr_path} already exists, choose another name"
-
-        zarr_root = zarr.open(zarr_path_init, mode="w", storage_options=storage_options)
-
-        if self.check_if_json_serializable():
-            zarr_root.attrs["provenance"] = check_json(self.to_dict())
-        else:
-            zarr_root.attrs["provenance"] = None
-
-        # save data (done the subclass)
-        save_kwargs["zarr_root"] = zarr_root
         save_kwargs["zarr_path"] = zarr_path
         save_kwargs["storage_options"] = storage_options
         save_kwargs["channel_chunk_size"] = channel_chunk_size
         cached = self._save(verbose=verbose, **save_kwargs)
```

Comment:
Should be:
```suggestion
        cached = self._save(format="zarr", verbose=verbose, **save_kwargs)
```
no ?

Reply:
I think it's already in the `save_kwargs`